lib/mix/tasks/ash.gen.resource.ex

```diff
@@ -0,0 +1,235 @@
+      end
+
+      #{unless no_code_interface?, do: code_interface_content}
+    end
+    """
+
+    api_dir_path = "lib/#{String.replace(api_module_name, ".", "/") |> String.downcase()}"
+    resource_dir_path = "#{api_dir_path}/resources"
+
+    # Ensure directories exist
+    File.mkdir_p!(api_dir_path)
+    File.mkdir_p!(resource_dir_path)
+
+    api_file_path = "#{api_dir_path}.ex"
+    resource_file_path = "#{resource_dir_path}/#{String.downcase(resource_name)}.ex"
+
+    if File.exists?(api_file_path) do
+      content = File.read!(api_file_path)
+
+      unless String.contains?(content, "resource #{resource_module_name}") do
+        # Append the new resource to the existing API file
+        parts = String.split(content, "\n")
+        index = Enum.find_index(parts, fn line -> String.trim(line) == "end" end)
+
+        if index do
+          parts = List.insert_at(parts, index, "    resource #{resource_module_name}")
+          File.write!(api_file_path, Enum.join(parts, "\n"))
+          IO.puts("Modified #{api_file_path}")
+        end
+      end
+    else
+      File.write!(api_file_path, """
+      defmodule #{api_module_name} do
+        use Ash.Api
+
+        resources do
+          resource #{resource_module_name}
+        end
+      end
+      """)
+
+      IO.puts("Generated #{api_file_path}")
+    end
+
+    # Check if the API is listed in the configuration of the current application
+    unless api_listed?(api_module_name) do
+      IO.puts("""
+      \e[31m
+      WARNING:
+      #{api_module_name} is not yet listed in the application's configuration.\e[0m
+      Make sure to add it to the configuration. Here is an example:
+
+      ** config/config.exs **
+      import Config
+
+      config :#{Macro.underscore(app_name)}, :ash_apis, [#{api_module_name}]
+
+      """)
+    end
+
+    if File.exists?(resource_file_path) do
+      IO.puts("File #{resource_file_path} already exists. Replace? [Yn]")
+      response = IO.gets("> ") |> String.trim()
+
+      if String.downcase(response) not in ["y", "yes", ""] do
+        IO.puts("Aborted!")
+        exit(:normal)
+      end
+    end
+
+    File.write!(resource_file_path, resource_file_content)
+    IO.puts("Generated #{resource_file_path}")
+
+    if uses_ash_postgres?() do
+      IO.puts("\nYou may need to run 'mix ash.codegen' and 'mix ash_postgres.migrate'.")
+    end
+
+    # Format the generated resource
+    Mix.Tasks.Format.run([api_file_path, resource_file_path])
+  end
+
+  defp app_name do
+    app_name_atom = Mix.Project.config()[:app]
+    Macro.camelize(Atom.to_string(app_name_atom))
+  end
+
+  defp uses_ash_postgres? do
+    mix_file = "mix.exs"
+    content = File.read!(mix_file)
+
+    # Regex to find ash_postgres in deps
+    # This approach has limitations, and there are many ways in which deps can be defined,
+    # but it should work for standard dep definitions.
+    Regex.match?(~r/ash_postgres/, content)
```

Comment:
I'd use `Code.ensure_loaded?(AshPostgres)` as opposed to reading the mix file. Also, I've thought about this a lot and I think we shouldn't make assumptions like this, but rather make it interactive. i.e "choose a data layer, here are the available data layers", that kind of thing.

Reply:
and support `--data-layer AshPostgres.DataLayer` otherwise.